### ISSUE TITLE
Can't start cluster somehow

### ISSUE THREAD

**Issue by jianpan** [jtriley/StarCluster#278]
I heard about starcluster from pydata 2013. It seems to be a nice tool to simplify managing EC2 cluster from python. However I'm getting the following error following the getting start guide. Can someone give me a hint what might be wrong? Thanks.

```
StarCluster - (http://star.mit.edu/cluster) (v. 0.94)
Software Tools for Academics and Researchers (STAR)
Please submit bug reports to starcluster@mit.edu

>>> Using default cluster template: smallcluster
>>> Validating cluster template settings...
>>> Cluster template settings are valid
>>> Starting cluster...
>>> Launching a 2-node cluster...
>>> Creating security group @sc-mycluster2...
Reservation:r-d7a304b7
>>> Waiting for cluster to come up... (updating every 30s)
>>> Waiting for all nodes to be in a 'running' state...
2/2 |||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||| 100%  
!!! ERROR - No active cluster nodes found!
!!! ERROR - 
!!! ERROR - Below is a list of terminated instances:
!!! ERROR - 
!!! ERROR - i-0539856a (terminated) Server.InternalError: Internal error
!!! ERROR - on launch
!!! ERROR - i-0339856c (terminated) Server.InternalError: Internal error
!!! ERROR - on launch
```

**Comment by jtriley** [comment]
@jianpan Sorry you're having issues. I'm not quite sure why you're getting an `internal error on launch`. What instance type and AMI are you using? Also what region are you in? I'll test this and see if I can reproduce the issue...

**Comment by jtriley** [comment]
@jianpan Can you also post the output of `$ starcluster listclusters` after getting this failure?

**Comment by jianpan** [comment]
I tried the git version and it worked!

It seems the pip installed version has some problem.

The configuration I'm using are mostly defaults. ie. us-east-1, x1.small, 2 node cluster etc. I used pip install instead of easy_install. Could that have some compatibility problem?

The listcluster command shows something like this:

```
StarCluster - (http://star.mit.edu/cluster) (v. 0.94)
Software Tools for Academics and Researchers (STAR)
Please submit bug reports to starcluster@mit.edu

-----------------------------------------
mycluster (security group: @sc-mycluster)
-----------------------------------------
Launch time: N/A
Uptime: N/A
Zone: N/A
Keypair: N/A
EBS volumes: N/A
Cluster nodes: N/A
```

terminate command will complain no master node and will not remove mycluster. But other clustered I started work fine now with the latest version from git.

**Comment by jtriley** [comment]
@jianpan The git version only has a couple of commits beyond the 0.94 release so they should be mostly identical. At this point I'm wondering if this is a transient error that disappeared with time rather than an issue with installing using pip. Also, the terminate command recommends using the -f option when there's no master node to force termination:

```
$ starcluster terminate -f mycluster2
```

**Comment by jianpan** [comment]
Thanks, Justin.

**Comment by jtriley** [comment]
@jianpan No problem. If you wouldn't mind could you try to install the latest stable 0.94 with pip again and see if you can reproduce this? I'm curious whether this is a random EC2 failure or an issue in the code. Thanks!

**Comment by jianpan** [comment]
@jtriley I just tried pip install on another box, and everything worked at first try. I guess it's just some random hiccup this afternoon. Very excited at starcluster!

**Comment by jtriley** [comment]
@jianpan Ugh that's bad luck for StarCluster :P Thanks for hanging in there and testing it again!

**Comment by rcabreu** [comment]
I was having the same issue. I changed the instance type and it worked perfectly! :-) Great tool.

**Comment by pieper** [comment]
I had the same issue trying to do the smallcluster sample - it failed three times in a row, and then it worked but only the master node started.  node001 terminated during startup.

**Comment by yunmingzhang17** [comment]
I had the same issue and it won't go away with multiple retrys. I was using the intact original configuration for small cluster (of course, I changed the key name)

I ended up switching to t1.micro instances from m1.small(default) and it seems to work now.

**Comment by jtriley** [comment]
@rcabreu @pieper @yunmingzhang17 Does this issue still exist for you in 0.94.2?

**Comment by aginpatrick** [comment]
Same thing for me. Original configuration for smallcluster. StarCluster version 0.94.2. Three failures in a row. I switched to t1.micro from m1.small. It works. Thanks Justin for this great tool!

**Comment by epowers** [comment]
OSX 10.7 easy_install StarCluster, 0.94.2, removed ~/.starcluster for a fresh start, only changed the three lines for AWS: ACCESS..., SECRET..., USER_ID

bash-3.2$ starcluster start mycluster
StarCluster - (http://star.mit.edu/cluster) (v. 0.94.2)
Software Tools for Academics and Researchers (STAR)
Please submit bug reports to starcluster@mit.edu

**\* WARNING - Setting 'EC2_PRIVATE_KEY' from environment...
**\* WARNING - Setting 'AWS_SECRET_ACCESS_KEY' from environment...
**\* WARNING - Setting 'EC2_CERT' from environment...
**\* WARNING - Setting 'AWS_ACCESS_KEY_ID' from environment...

> > > Using default cluster template: smallcluster
> > > Validating cluster template settings...
> > > Cluster template settings are valid
> > > Starting cluster...
> > > Launching a 2-node cluster...
> > > Creating security group @sc-mycluster...
> > > Reservation:r-f5eb9191
> > > Waiting for instances to propagate... 
> > > Waiting for cluster to come up... (updating every 30s)
> > > Waiting for all nodes to be in a 'running' state...
> > > 1/1 |||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||| 100%  
> > > !!! ERROR - No active cluster nodes found!
> > > !!! ERROR - 
> > > !!! ERROR - Below is a list of terminated instances:
> > > !!! ERROR - 
> > > !!! ERROR - i-544dc130 (terminated) Server.InternalError: Internal error
> > > !!! ERROR - on launch
> > > !!! ERROR - i-564dc132 (terminated) Server.InternalError: Internal error
> > > !!! ERROR - on launch

# 

Changing instance type to t1.micro works.

# 

I reinstalled with pip, did not fix the problem.

sudo pip uninstall StarCluster
sudo pip install StarCluster
starcluster help
vi ~/.starcluster/config

# change credentials

bash-3.2$ starcluster start mycluster
StarCluster - (http://star.mit.edu/cluster) (v. 0.94.2)
Software Tools for Academics and Researchers (STAR)
Please submit bug reports to starcluster@mit.edu

**\* WARNING - Setting 'EC2_PRIVATE_KEY' from environment...
**\* WARNING - Setting 'AWS_SECRET_ACCESS_KEY' from environment...
**\* WARNING - Setting 'EC2_CERT' from environment...
**\* WARNING - Setting 'AWS_ACCESS_KEY_ID' from environment...

> > > Using default cluster template: smallcluster
> > > Validating cluster template settings...
> > > Cluster template settings are valid
> > > Starting cluster...
> > > Launching a 2-node cluster...
> > > Creating security group @sc-mycluster...
> > > Reservation:r-3e2e4259
> > > Waiting for instances to propagate... 
> > > Waiting for cluster to come up... (updating every 30s)
> > > Waiting for all nodes to be in a 'running' state...
> > > 1/1 |||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||| 100%  
> > > !!! ERROR - No active cluster nodes found!
> > > !!! ERROR - 
> > > !!! ERROR - Below is a list of terminated instances:
> > > !!! ERROR - 
> > > !!! ERROR - i-544dc130 (terminated) Server.InternalError: Internal error
> > > !!! ERROR - on launch
> > > !!! ERROR - i-564dc132 (terminated) Server.InternalError: Internal error
> > > !!! ERROR - on launch
> > > !!! ERROR - i-1241af74 (terminated) Server.InternalError: Internal error
> > > !!! ERROR - on launch
> > > !!! ERROR - i-1041af76 (terminated) Server.InternalError: Internal error
> > > !!! ERROR - on launch
> > > !!! ERROR - i-a3a435db (terminated) Client.UserInitiatedShutdown: User
> > > !!! ERROR - initiated shutdown
> > > !!! ERROR - i-a5a435dd (terminated) Client.UserInitiatedShutdown: User
> > > !!! ERROR - initiated shutdown

# 

Checked out github master v 0.9999, 3d4471e7096bdee87e450f53b92aedf088e6b6af, built, installed, same problem.

# 

Changed the AMI from the 32-bit to the 64-bit (ami-765b3e1f) and tried m1.small and m1.large, both worked!

I recommend changing the default starcluster config to either use t1.micro as the default or use the 64-bit AMI as the default.

**Comment by jtriley** [comment]
@aginpatrick @epowers Thanks for following up.

This is _all_ fixed now by gh-329 in commits d6eb8934 and 1d508370

Turns out there was an issue with how StarCluster customizes the block device map when using m1.small with a 32bit AMI - no other combination of instance type/AMI seems to cause the same problem. To make matters worse the default AMI in the config template is a 32bit AMI and the default instance type is m1.small. I've fixed the block device map issue but decided to go ahead and change the default AMI to the 64bit AMI anyway as @epowers suggested given that 64bit is the most flexible arch on EC2 these days.
